lightly/api/__init__.py

```diff
@@ -5,3 +5,10 @@
 
 from lightly.api.api_workflow_client import ApiWorkflowClient
 from lightly.api.api_workflow_compute_worker import ArtifactNotExist
+
+# Mock the urlencode to use doseq, so that arrays in the query are properly encoded
+from six.moves.urllib.parse import urlencode
+def doseq_urlencode(fields):
+    return urlencode(fields, doseq=True)
+import urllib3.request
+urllib3.request.urlencode = doseq_urlencode
```

Comment:
This is pretty hacky as it changes urllib3 under the hood. A less intrusive alternative is to patch the requests method on the rest client:
```
from lightly.openapi_generated.swagger_client.rest import RESTClientObject

def patch_rest_client(rest_client):
    request = rest_client.request

    def request_patched(self, method, url, query_params=None, headers=None,
                    body=None, post_params=None, _preload_content=True,
                    _request_timeout=None):
        if query_params is not None:
            new_query_params = []
            for name, value in query_params:
                if isinstance(value, list):
                    new_query_params.extend([(name, val) for val in value])
                else:
                    new_query_params.append((name, value))
            query_params = new_query_params
        return request(self, method=method, url=url, query_params=query_params, headers=headers, body=body, post_params=post_params, _preload_content=_preload_content, _request_timeout=_request_timeout)

    rest_client.request = request_patched

patch_rest_client(RESTClientObject)
```

What it does is convert query parameters that look like this: `[('labels', ['A', 'B', 'C'])]` into a list of simple parameters: `[('labels', 'A'), ('labels', 'B'), ('labels', 'C')]` which is supported by swagger codegen.

Reply:
Thanks, good idea!
However, this still messes up with a library, but luckily only a generated one.